pkgs/by-name/g-/g-golf/package.nix

```
@@ -0,0 +1,86 @@
+{
+  stdenv,
+  lib,
+  fetchurl,
+  pkg-config,
+  automake117x,
+  autoconf,
+  libtool,
+  gnumake,
+  gcc,
+  gnum4,
+  gettext,
+  texinfo,
+  guile,
+  guile-lib,
+  gobject-introspection,
+  glib,
+  xvfb-run,
+  gtk3,
+  gtk4,
+  which,
+  libadwaita,
+  ...
+}:
+stdenv.mkDerivation (finalAttrs: {
+
+  pname = "g-golf";
+  version = "0.8.1";
+
+  src = fetchurl {
+    url = "http://ftp.gnu.org/gnu/g-golf/g-golf-${finalAttrs.version}.tar.gz";
+    hash = "sha256-yl5pSENYcOqx7TUWmzSXHhj02HuCr+ak4UsTtDIxjyI=";
+  };
+
+  doCheck = true;
+  makeFlags = [ "GUILE_AUTO_COMPILE=0" ];
+
+  nativeBuildInputs = [
+    pkg-config
+    automake117x
+    autoconf
+    libtool
+    gnumake
+    gcc
+    gnum4
+    gettext
+    texinfo
+    guile
+    guile-lib
+    xvfb-run
+    gtk3
+  ];
+
+  buildInputs = [
+    glib.out
+    gobject-introspection
+  ];
+
+  postPatch = ''
+    sed -i 's|libgirepository-1.0|${gobject-introspection}/lib/libgirepository-1.0.so|' g-golf/init.scm
+    sed -i "s|libglib-2.0|${glib.out}/lib/libglib-2.0.so|" g-golf/init.scm
+    sed -i "s|libgobject-2.0|${glib.out}/lib/libgobject-2.0.so|" g-golf/init.scm
+
+    # In build environment, There is no /dev/tty
+    sed -i "s|/dev/tty|/dev/null|g" test-suite/tests/gobject.scm
+
+    sed -i "s|SITEDIR=.*$|SITEDIR="$out/${guile.siteDir}"|" configure.ac
+    sed -i "s|SITECCACHEDIR=.*$|SITECCACHEDIR="$out/${guile.siteCcacheDir}"|" configure.ac
+  '';
+
+  checkPhase = ''
+    xvfb-run make check
+  '';
```

```suggestion
  checkPhase = ''
    runHook preCheck
    xvfb-run make check
    runHook postCheck
  '';
```
Do not omit calling hooks, so that others could easily override your package